lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -340,14 +335,10 @@ def __init__(self, canvas, num):
         self.set_window_title("Figure %d" % num)
         try:
             self.window.set_icon_from_file(window_icon)
-        except (SystemExit, KeyboardInterrupt):
```

Comment:
Why are these not being re-raised anymore?

Reply:
We now catch Exception, which is everything *but* SystemExit, KeyboardInterrupt (and GeneratorExit, which is irrelevant here).  SystemExit and KeyboardInterrupt are not caught at all and just get raised normally.